### PR TITLE
Add Azure container instance net SDK

### PR DESCRIPTION
**Pull request by yolocs**
## Description

This is the **first** version of [Azure container instance](https://azure.microsoft.com/en-us/services/container-instances/) .net SDK.

The API spec has already been reviewed and merged. See below.

- [Container Instance REST API spec](https://github.com/Azure/azure-rest-api-specs/tree/current/specification/containerinstance/resource-manager)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerInstance.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+    <StartupObject></StartupObject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- When xunit cannot detect tests, uncomment the following dependencies -->
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerInstance.Tests.csproj]: @yolo3301 remove commented code

File: src/SDKs/ContainerInstance/Management.ContainerInstance/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure Container Instance Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Container Instance.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.2.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/Management.ContainerInstance/Properties/AssemblyInfo.cs]: @yolo3301 have consistent versions between nuget version as well as AssemblyFile Version.

File: src/SDKs/dirs.proj [hunk 3]
```diff
@@ -11,6 +11,7 @@
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Cdn\Management.Cdn\Microsoft.Azure.Management.Cdn.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)CognitiveServices\Management.CognitiveServices\Microsoft.Azure.Management.CognitiveServices.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Compute\Management.Compute\Microsoft.Azure.Management.Compute.csproj" />
+		<ProjectToBuild Include="$(MSBuildThisFileDirectory)ContainerInstance\Management.ContainerInstance\Microsoft.Azure.Management.ContainerInstance.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/dirs.proj]: @yolo3301 dirs.proj is no longer used, you can revert this change.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@yolo3301 please provide detailed description about this PR.
If this is your first version, please specify it else describe the changes going in this version.

**Comment by brendandburns** [comment]
awesome, thank you!

**Comment by yolo3301** [comment]
@shahabhijeet Thanks for reviewing. All comments have been addressed.

File: src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerInstance.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  <PropertyGroup>
+    <PackageId>ContainerInstance.Tests</PackageId>
+    <Description>ContainerInstance.Tests Class Library</Description>
+    <AssemblyName>ContainerInstance.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
+    <StartupObject></StartupObject>
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerInstance.Tests.csproj]: @yolo3301 please remove this

File: src/SDKs/ContainerInstance/ContainerInstance.Tests/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("ContainerInstance.Tests")]
+[assembly: AssemblyCopyright("Copyright ©  2016")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible 
+// to COM components.  If you need to access a type in this assembly from 
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/ContainerInstance.Tests/Properties/AssemblyInfo.cs]: nit:
@yolo3301 you made this assembly COM visible false, but then providing typelib GUID
Please consider removing this.

File: src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <Description>Microsoft Azure Container Instance Management Library</Description>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <AssemblyName>Microsoft.Azure.Management.ContainerInstance</AssemblyName>
+    <PackageId>Microsoft.Azure.Management.ContainerInstance</PackageId>
+    <PackageTags>Microsoft Azure Container Instance Management;Container Instance;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288890</PackageIconUrl>
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj]: @yolo3301 please remove this tag, as it's centrally controlled.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@yolo3301 this is your first time, are you sure you want to go 1.0.0 or you rather go 1.0.0-preview
If so, please mark as that version in csproj file and not AssemblyInfo.cs

**Comment by yolo3301** [comment]
@shahabhijeet I do want to go with 1.0.0-preview. So I removed the versions in AssemblyInfo.cs.

**Review by shahabhijeet — APPROVED** [review]
as soon as CI passes, this is good to go

**Comment by yolo3301** [comment]
@shahabhijeet The other check failed. However, I don't have permission to check what's wrong there. Could you help?

**Pull request merged**